SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -3531,6 +3533,136 @@ bool USpatialNetDriver::HasTimedOut(const float Interval, uint64& TimeStamp)
 	return false;
 }
 
+void USpatialNetDriver::ServerReplicateActors_BuildConsiderList(TArray<FNetworkObjectInfo*>& OutConsiderList, const float ServerTickTime)
+{
+	TArray<FNetworkObjectInfo*> TmpConsiderList;
+	TmpConsiderList.Reserve(GetNetworkObjectList().GetActiveObjects().Num());
+
+	Super::ServerReplicateActors_BuildConsiderList(TmpConsiderList, ServerTickTime);
+
+	// Cache if we are running with Spatial Networking to avoid getting the settings in every iteration of the loop.
+	const bool bSpatialNetworking = GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking();
+	if (!bSpatialNetworking)
+	{
+		return;
+	}
+
+	for (FNetworkObjectInfo* const& ActorInfo : TmpConsiderList)
+	{
+		AActor* Actor = ActorInfo->Actor;
+		check(Actor != nullptr);
+
+		Worker_EntityId EntityId = PackageMap->GetEntityIdFromObject(Actor);
+
+		// Check for unauthorised data changes no non-auth actors and add auth actors only to the consider list
+		if (!Actor->HasAuthority())
+		{
+			CheckUnauthorisedDataChanges(EntityId, Actor);
+		}
+		else
+		{
+			OutConsiderList.Add(ActorInfo);
+		}
+	}
+}
+
+void USpatialNetDriver::CheckUnauthorisedDataChanges(Worker_EntityId_Key EntityId, AActor* Actor)
+{
+	if (!IsServer())
+	{
+		return;
+	}
+
+	if (!SpatialShadowActors.Contains(EntityId))
+	{
+		// Haven't received an update yet for this entity so can't check
+		return;
+	}
+
+	if (SpatialShadowActors[EntityId] != nullptr)
```

Comment:
do you want only a null check here, or do you want a variation of IsValid()?

Reply:
Good point used IsValid check instead